### ISSUE TITLE
[Feature Request] Support for Yeelight devices

### ISSUE THREAD

**Issue by slicciardello** [Domi04151309/HomeApp#3]
I use several Yeelight devices, they are widely used devices. The proprietary application does not seem very privacy-friendly... It would be really great if they were supported by this application! 

There are several repos available on github, maybe some can be useful
https://github.com/topics/yeelight

**Comment by Domi04151309** [comment]
I found a [python script](https://github.com/aluminiumgeek/goodbye-mihome/blob/master/plugins/yeelight.py) that looks very promising and a [Java API](https://github.com/florian-mollin/yapi) that could be useful. The Java API is pretty old but it should be possible to rewrite or implement it.

**Comment by Domi04151309** [comment]
Also, a detailed specification can be found on the [Yeelight website](https://www.yeelight.com/download/Yeelight_Inter-Operation_Spec.pdf).

**Comment by slicciardello** [comment]
> I found a [python script](https://github.com/aluminiumgeek/goodbye-mihome/blob/master/plugins/yeelight.py) that looks very promising and a [Java API](https://github.com/florian-mollin/yapi) that could be useful. The Java API is pretty old but it should be possible to rewrite or implement it.

Nice, I can try if they work if you can implement them into the application. When you have test builds tell me and I'll try them right away.

My devices where I can test are:
[Mi Led Desk Lamp](https://www.mi.com/global/smartlamp)
[Yeelight LED Ceiling Light](https://www.yeelight.com/en_US/product/luna)
If there's something I need to do by code, tell me what to do step by step, I'm still not very good at it.

**Comment by Domi04151309** [comment]
I have implemented some basic functionality now and would need you to test it. Do you know how to build the application from source using Android Studio?

**Comment by slicciardello** [comment]
> I have implemented some basic functionality now and would need you to test it. Do you know how to build the application from source using Android Studio?

I've never tried, if you show me a guide on how to do it I can try. I use linux

**Comment by Domi04151309** [comment]
#### Long Version
First, you have to install [Android Studio](https://developer.android.com/studio) and download the source from the [`yeelight` branch](https://github.com/Domi04151309/HomeApp/archive/yeelight.zip). After you have done that you should be able to open the project using Android Studio. There you have to select your device in the top bar and allow debugging from this computer if you haven't done already. Then you can click the "Run" button (green play button) and it should build and install the app.

#### Short Version
- Install Android Studio
- Clone the yeelight branch
- Open the project in Android Studio
- Connect your device and allow debugging
- Build and install the app

#### Things That Should Work
- On/Off buttons
- Changing properties after releasing the sliders

**Comment by slicciardello** [comment]
> #### Long Version
> 
> First, you have to install [Android Studio](https://developer.android.com/studio) and download the source from the [`yeelight` branch](https://github.com/Domi04151309/HomeApp/archive/yeelight.zip). After you have done that you should be able to open the project using Android Studio. There you have to select your device in the top bar and allow debugging from this computer if you haven't done already. Then you can click the "Run" button (green play button) and it should build and install the app.
> #### Short Version
> 
>     * Install Android Studio
> 
>     * Clone the yeelight branch
> 
>     * Open the project in Android Studio
> 
>     * Connect your device and allow debugging
> 
>     * Build and install the app
> 
> 
> #### Things That Should Work
> 
>     * On/Off buttons
> 
>     * Changing properties after releasing the sliders

Thanks i will try soon

**Comment by slicciardello** [comment]
> #### Long Version
> 
> First, you have to install [Android Studio](https://developer.android.com/studio) and download the source from the [`yeelight` branch](https://github.com/Domi04151309/HomeApp/archive/yeelight.zip). After you have done that you should be able to open the project using Android Studio. There you have to select your device in the top bar and allow debugging from this computer if you haven't done already. Then you can click the "Run" button (green play button) and it should build and install the app.
> #### Short Version
> 
>     * Install Android Studio
> 
>     * Clone the yeelight branch
> 
>     * Open the project in Android Studio
> 
>     * Connect your device and allow debugging
> 
>     * Build and install the app
> 
> 
> #### Things That Should Work
> 
>     * On/Off buttons
> 
>     * Changing properties after releasing the sliders
![image](https://user-images.githubusercontent.com/65731761/83751650-76036e00-a667-11ea-8fa4-975934769d70.png)

**Comment by Domi04151309** [comment]
> ![image](https://user-images.githubusercontent.com/65731761/83751650-76036e00-a667-11ea-8fa4-975934769d70.png)

https://stackoverflow.com/questions/54273412/failed-to-install-the-following-android-sdk-packages-as-some-licences-have-not

**Comment by slicciardello** [comment]
> > ![image](https://user-images.githubusercontent.com/65731761/83751650-76036e00-a667-11ea-8fa4-975934769d70.png)
> 
> https://stackoverflow.com/questions/54273412/failed-to-install-the-following-android-sdk-packages-as-some-licences-have-not

Thanks now works. But i tried to add Mi Lamp, it can't connect.
![image](https://user-images.githubusercontent.com/65731761/83754403-32f7c980-a66c-11ea-90a6-c37dcdab20dc.png)
![image](https://user-images.githubusercontent.com/65731761/83754417-368b5080-a66c-11ea-9b47-a199fb4d5475.png)
![image](https://user-images.githubusercontent.com/65731761/83754426-38edaa80-a66c-11ea-8e92-c20515ce1e68.png)
![image](https://user-images.githubusercontent.com/65731761/83754434-3be89b00-a66c-11ea-8be8-fd7618d715bb.png)

**Comment by Domi04151309** [comment]
Have you tried pressing the buttons and using the sliders

**Comment by slicciardello** [comment]
> Have you tried pressing the buttons and using the sliders?

yes, they didn't work. It is a connection problem. I must unistall yeelight application first?

**Comment by Domi04151309** [comment]
No I think you have to enable a setting in the yeelight app.

**Comment by slicciardello** [comment]
> No I think you have to enable a setting in the yeelight app.

No setting to enable.. Maybe i need to reset lamp, pressing is reset button?

**Comment by Domi04151309** [comment]
Maybe the library doesn't work or is misconfigured

**Comment by slicciardello** [comment]
> Maybe the library doesn't work or is misconfigured

Pressing reset button address change?

**Comment by Domi04151309** [comment]
Oops accidentally closed the issue.

No I think that the implementation isn't correct.

**Comment by slicciardello** [comment]
> ![image](https://user-images.githubusercontent.com/65731761/83771027-18c8e600-a682-11ea-8de2-ab7bfc3c1fed.png)
> 
> It asked me to enable location and bluetooth for paring after resetting the lamp. I think that your implementation is correct for some bulb, but not for newer desk lamp and other. I will try my ceiling when i can mount hit.

Maybe it need mac adress and not ip adress?

**Comment by Domi04151309** [comment]
It needs the IP address

**Comment by slicciardello** [comment]
> It needs the IP address

Ok, let me know when there's any news

**Comment by trmendes** [comment]
I'm trying this branch but it does not connect to the device.

**Comment by Domi04151309** [comment]
That's because the implementation is incorrect. It doesn't work yet.

**Comment by trmendes** [comment]
I see.
Thanks.
Please let me know in case you need any help with testing
